File: BE/src/main/java/com/application/core/shared/entities/Address.java
package com.application.core.shared.entities;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)

public class Address {
  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  private Long id;
  private String address;
  private String addressArabic;
  private String city;

  @Column(columnDefinition = "tinyint(1) default 1")
  private boolean active;



  public JsonNode toJson() {
    ObjectMapper objectMapper = new ObjectMapper();
    return objectMapper.valueToTree(this);
  }
}

File: FE/src/app/modules/shared/model/address.ts
export class Address {
  private _id: number;
  private _address: string;
  private _addressArabic: string;
  private _city: string;
  private _active: boolean;



  get id(): number {
    return this._id;
  }
  set id(value: number) {
    this._id = value;
  }

  get address(): string {
    return this._address;
  }
  set address(value: string) {
    this._address = value;
  }

  get addressArabic(): string {
    return this._addressArabic;
  }
  set addressArabic(value: string) {
    this._addressArabic = value;
  }

  get city(): string {
    return this._city;
  }
  set city(value: string) {
    this._city = value;
  }

  get active(): boolean {
    return this._active;
  }
  set active(value: boolean) {
    this._active = value;
  }



  get toJson(): any {
    return {
      id: this.id ? this.id : null,
      address: this.address,
      addressArabic: this.addressArabic,
      city: this.city,
      active: this.active ?? true
    };
  }

  static fromJson(json: any): Address {
    const address = new Address();
    address.id = json.id;
    address.address = json.address;
    address.addressArabic = json.addressArabic;
    address.city = json.city;
    address.active = json.active;

    return address;
  }
}
